packages/text-field/Input.js

```diff
@@ -96,7 +119,11 @@ export default class Input extends React.Component {
   // value of the input is.
   handleChange = (e) => {
     const {foundation, onChange} = this.props;
+    // autoCompleteFocus runs on `input` event in MDC Web. In React, onChange and
+    // onInput are the same event
+    // https://stackoverflow.com/questions/38256332/in-react-whats-the-difference-between-onchange-and-oninput
     foundation.autoCompleteFocus();
+    this.setState({disableProgramaticSetValue: true});
```

Comment:
So the `onChange` event is only emitted when it's from a user? Do programmatic changes emit any event?

Reply:
Right the onChange event is only triggered when a user types. BOTH events (user and programatic) trigger updates props, which calls `componentDidUpdate`.